### PR TITLE
Enable DRC related components on TGL override config

### PR DESCRIPTION
**Pull request by johnylin76**
Enabled CROSSOVER, DRC, and MULTIBAND_DRC component configs to tigerlake_chrome.config in order to deploy DRC/EQ on speaker pipeline on TGL device. We will first turn on DRC on Eldrid.

### CODE REVIEW

File: src/audio/Kconfig [hunk 1]
```diff
@@ -172,6 +172,19 @@ config COMP_DRC
 	  to reduce the volume of loud sounds and amplify silent sounds thus
 	  compressing an audio signal's dynamic range.
 
+if COMP_DRC
+
+config DRC_MAX_PRE_DELAY_FRAMES
+	int "DRC max number of pre-delay frames"
+	default 256
```
Comment (cujomalainey) [inline src/audio/Kconfig:179]: i think you dont need the if, just add "depends on"

File: src/audio/drc/drc.c [hunk 2]
```diff
@@ -68,7 +68,7 @@ inline int drc_init_pre_delay_buffers(struct drc_state *state,
 				      size_t sample_bytes,
 				      int channels)
 {
-	size_t bytes_per_channel = sample_bytes * DRC_MAX_PRE_DELAY_FRAMES;
+	size_t bytes_per_channel = sample_bytes * CONFIG_DRC_MAX_PRE_DELAY_FRAMES;
```
Comment (cujomalainey) [inline src/audio/drc/drc.c:71]: have you tried with rballoc? That might be way you are hitting ENOMEM

**Review by cujomalainey — CHANGES_REQUESTED** [review]

File: src/audio/drc/drc.c (hunk 2 above)
Comment (johnylin76) [reply src/audio/drc/drc.c:71]: You are absolutely right!! I just knew that there are several memory heap zones to allocate from.
https://thesofproject.github.io/latest/api/memory-alloc-api.html

Comment (johnylin76) [reply src/audio/drc/drc.c:71]: Then I changed the default value of DRC_MAX_PRE_DELAY_FRAMES to 512. The pre-delay second is 0.006 in CRAS and the correspondent frame number is 288 (0.006*48000). The max frame number just needs to be larger than 288 and be a 2^N number, then we could guarantee the behavior is coherent as CRAS.

File: src/include/sof/audio/drc/drc.h [hunk 3]
```diff
@@ -15,10 +15,12 @@
 struct audio_stream;
 struct comp_dev;
 
-/* DRC_MAX_PRE_DELAY_FRAMES needs to be a 2^N number */
-#define DRC_MAX_PRE_DELAY_FRAMES 1024
-#define DRC_MAX_PRE_DELAY_FRAMES_MASK (DRC_MAX_PRE_DELAY_FRAMES - 1)
-#define DRC_DEFAULT_PRE_DELAY_FRAMES 256
+#ifndef CONFIG_DRC_MAX_PRE_DELAY_FRAMES
```
Comment (lgirdwood) [inline src/include/sof/audio/drc/drc.h:18]: Is this needed since the Kconfig value will be mandatory on all builds and be 512 by default ?

Comment (johnylin76) [reply src/include/sof/audio/drc/drc.h:18]: I think it would be needed for testbench purpose. e.g. ./scripts/rebuild-testbench.sh

Comment (lgirdwood) [reply src/include/sof/audio/drc/drc.h:18]: ah, good point. we need to think about how test bench can use Kconfig....

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI is showing known and unrelated jack detect error.

File: src/include/sof/audio/drc/drc.h (hunk 3 above)
Comment (cujomalainey) [reply src/include/sof/audio/drc/drc.h:18]: can you leaver a comment/todo so when we fix it we can remove it? because it does look weird. @marc-hb FYI

Comment (marc-hb) [reply src/include/sof/audio/drc/drc.h:18]: How about:
```suggestion
#ifdef CONFIG_LIBRARY
#define CONFIG_DRC_MAX_PRE_DELAY_FRAMES 512
#endif
```
This makes sure firmware compilation fails if someone breaks the new Kconfig above. Plus: self-commenting code :-)

**Review by lgirdwood — APPROVED** [review]
@cujomalainey pls merge if you approve.

**Review by cujomalainey — APPROVED** [review]

**Comment by cujomalainey** [comment]
Failure on unchanged platform

**Pull request merged**